src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -70,15 +68,35 @@ template <typename T>
 struct RadixContainer {
   std::shared_ptr<Partition<T>> elements;
   std::vector<size_t> partition_offsets;
+
+  // bit vector to store NULL flags
+  std::shared_ptr<std::vector<bool>> position_is_null_value;
 };
 
-template <typename T, typename HashedType>
+inline std::vector<size_t> determine_chunk_offsets(std::shared_ptr<const Table> table) {
+  const size_t chunk_count = table->chunk_count();
+  auto chunk_offsets = std::vector<size_t>(chunk_count);
+
+  size_t offset = 0;
+  for (ChunkID chunk_id{0}; chunk_id < chunk_count; ++chunk_id) {
+    chunk_offsets[chunk_id] = offset;
+    offset += table->get_chunk(chunk_id)->size();
+  }
+  return chunk_offsets;
+}
+
+template <typename T, typename HashedType, bool consider_null_values = false>
```

Comment:
Did you check if this has any performance implications? Remember that using this as a template parameter as opposed to a regular parameter doubles the compile time.

Reply:
Will do.